src/util/psbt/map/input.rs

```diff
@@ -264,8 +264,8 @@ where
             let val: Vec<u8> = Deserialize::deserialize(&raw_value)?;
             if <H as hashes::Hash>::hash(&val) != key_val {
                 return Err(psbt::Error::InvalidPreimageHashPair {
-                    preimage: val,
-                    hash: Vec::from(key_val.borrow()),
+                    preimage: val.into_boxed_slice(),
+                    hash: Vec::from(key_val.borrow()).into_boxed_slice(),
```

Comment:
It should be possible to directly do `Box::from(key_val.borrow())` or `Box::from(&key_val[..])`

Reply:
nice, should be fixed, thanks